include/engine/routing_algorithms/routing_base_mld.hpp

```diff
@@ -650,11 +650,38 @@ double getNetworkDistance(SearchEngineData<Algorithm> &engine_working_data,
         return std::numeric_limits<double>::max();
     }
 
-    std::vector<PathData> unpacked_path;
+    BOOST_ASSERT(unpacked_nodes.size() >= 1);
```

Comment:
I believe we could do similar optimisation for CH, but let's go step by step...

Reply:
For my understanding, there are two changes?
1. Remove unnecessary annotations irrelevant to distance calculation.
2. Use pre-calculated distances

Reply:
Yes, I think so

Reply:
Actually I think we can make much deeper optimization here: use 1 to N routing algorithm in map matching instead of running getNetworkDistance N times, but let's go with small steps...